### ISSUE TITLE
Missing strings in Transifex

### ISSUE THREAD

**Issue by Magnus26** [musescore/MuseScore#23298]
### Issue type

Other type of issue

### Bug description

In the Swedish (traditional) language version of MuseScore Studio the following strings from the dialog window "Lägg till eller ta bort instrument", Familj "Alla instrument",  Instrument "Fritunga", are missing in Transifex:

Diatoniskt munspel, 10 hål

Kromatiskt munspel, 12 hål

C Kromatiskt munspel, 14 hål

C Kromatiskt munspel, 16 hål

### Steps to reproduce

1. Open a score.
2. Click on the tab "Instrument".
3. Click on "Lägg till" to open the dialog window in question.

### MuseScore Version

4.3.2

### Regression

I don't know

### Operating system

Q4OS

**Comment by Jojo-Schmitz** [comment]
In the latest instruments_sv_SV.qm (from 'Update translations) nor on the corresponding 'original' (delivered with 4.3.2) I find any of these.
In the Add Instruments dialog, I can see that last 2:
![image](https://github.com/musescore/MuseScore/assets/1786669/9a4837bc-6ee2-4b9c-90b5-76da1505fc86)
Those 2 are also in the US-English version:
![image](https://github.com/musescore/MuseScore/assets/1786669/7b5593a2-639d-46a4-8cb9-7dc1fa8267ef)

There is, however, the String "C" and the strings "Kromatiskt munspel, 14 hål" and Kromatiskt munspel, 16 hål, whioch combined gove those 2 strings seen above, that shopuld explainn that part of the miracle.

No idea where and how to see the other 2 strings "Diatoniskt munspel, 10 hål" and "Kromatiskt munspel, 12 hål" though?

**Comment by cbjeukendrup** [comment]
Those strings are composed of [the string with context `engraving/instruments:harmonica-c14c trackName ` ("14-Hole Chromatic Harmonica")](https://app.transifex.com/musescore/musescore/translate/#sv/$/434662716) and [the string with context `engraving/instruments:harmonica-c14c traitName` ("*C")](https://app.transifex.com/musescore/musescore/translate/#sv/$/434342391). 

How these strings are combined is controlled by this string: https://app.transifex.com/musescore/musescore/translate/#sv/$/434016518

**Comment by cbjeukendrup** [comment]
> No idea where and how to see the other 2 strings "Diatoniskt munspel, 10 hål" and "Kromatiskt munspel, 12 hål" though?

https://app.transifex.com/musescore/musescore/translate/#sv/$/434662708?q=context%3Aharmonica and https://app.transifex.com/musescore/musescore/translate/#sv/$/434662710?q=context%3Aharmonica ? 

---

For more information, see https://github.com/musescore/MuseScore/wiki/Translating-instrument-names. 

I think this issue can be closed now, but in case of further questions, feel free to ask here.

**Comment by Magnus26** [comment]
> I think this issue can be closed now, but in case of further questions, feel free to ask here.

A problem is that "C Kromatiskt munspel" is bad Swedish. "Kromatiskt munspel i C" would be much better.

**Comment by cbjeukendrup** [comment]
Then you could translate [the `"%1 %2"` string](https://app.transifex.com/musescore/musescore/translate/#sv/$/434016518) with `"%2 i %1"`. That is the purpose of that string.

**Comment by Magnus26** [comment]
OK, I shall try that.

**Comment by Magnus26** [comment]
It wasn't totally successful: I got "Kromatiskt munspel, 14 hål i C" instead of "Kromatiskt munspel i C, 14 hål".

**Comment by cbjeukendrup** [comment]
Hm... @shoogle do you have any ideas about this? It looks like that would not be possible with the current system, and require a refinement. 

Maybe we should get rid of the idea of automatically composing strings. Doubtlessly, more problems like this will arise eventually. 
That would mean that translators have to translate more strings manually, but that might be overcomeable.

**Comment by Magnus26** [comment]
> Maybe we should get rid of the idea of automatically composing strings. Doubtlessly, more problems like this will arise eventually. That would mean that translators have to translate more strings manually, but that might be overcomeable.

I think it is better that you programmers work on other problems than automatic string composition, and let the translators compose the strings needed.

**Comment by shoogle** [comment]
> A problem is that "C Kromatiskt munspel" is bad Swedish. "Kromatiskt munspel i C" would be much better.

@Magnus26, maybe you know this already, but in English there is a semantic difference between "Horn in F" and "F Horn".

Name | Meaning
:---|:---
Horn in F | Horn that is transposing in the key of F (i.e. a written C sound like an F). We don't know its lowest note.
F Horn | Horn pitched in F (i.e. the lowest note it can play is an F). We don't known its transposition.

If there is a reasonable way to indicate this difference in Swedish then I would encourage you to use it.

> I got "Kromatiskt munspel, 14 hål i C" instead of "Kromatiskt munspel i C, 14 hål"

**TLDR:** The harmonicas are an anomaly. Don't use them to determine your naming scheme for the other instruments.

The harmonicas are unique in that their names have essentially 3 components:

- **Base name:** `Chromatic Harmonica`
- **Holes:** `14-Hole`
- **Pitch:** `C`

Most instruments only have 1 component to their name (e.g. `Violin`). Some have 2 components (e.g. `Horn` and `F`). Nothing has 3 components except the harmonicas.

Our automatic naming system can only cope with 1 or 2 components. @oktophonie and I discussed expanding it to allow for 3 components, but we decided it wasn't worth the added complexity given that the 3rd component would be empty for literally every instrument except the harmonicas.

Instead, we decided to combine the harmonica's base name and holes into one component (`14-Hole Chromatic Harmonica`). The other component (`C`) must appear either before that or after that; it can't go in the middle.

This might mean you need to use a non-ideal name like:

- `Kromatiskt munspel, 14 hål i C`
- `Kromatiskt munspel (14 hål) i C`
- `Kromatiskt munspel i C` (i.e. don't show `14 hål` in the score)
- `C Kromatiskt munspel, 14 hål`
- `14 hål kromatiskt munspel i C` 

We regard this as an acceptable compromise since the user is free to edit the instrument name if they don't like the default name.

In fact, most scores with harmonicas will only contain 1 harmonica and nothing else, so the instrument name would be hidden anyway. If there are other instruments, it's probably a piano or guitar, not another harmonica, so the user would edit the instrument name to say "Harmonica" and write the "C 14-hole Chromatic" bit somewhere else, like in a footnote or subtitle. You're unlikely to see "C 14-Hole Chromatic Harmonica" written out in full next to the stave in any published score.

> Maybe we should get rid of the idea of automatically composing strings.

We're not doing much with the name components at the moment, but in the future we'd like to give users more control over how instrument names appear in the score.

For example, in the **Format > Style** dialog, we may allow users to override the `%1 in %2` string. For example, the user might prefer to write `%1 (%2)` instead, which would make names like `Horn in F` appear as `Horn (F)`. This only works if the components are kept separate.

However, users would still have the option to set names explicitly if they prefer.

> Doubtlessly, more problems like this will arise eventually.

If that happens, please let me know!

**Comment by Magnus26** [comment]
> @Magnus26, maybe you know this already, but in English there is a semantic difference between "Horn in F" and "F Horn".
> Name 	Meaning
> Horn in F 	Horn that is transposing in the key of F (i.e. a written C sound like an F). We don't know its lowest note.
> F Horn 	Horn pitched in F (i.e. the lowest note it can play is an F). We don't known its transposition.
> 
> If there is a reasonable way to indicate this difference in Swedish then I would encourage you to use it.

In Swedish there is no such distinction: "B-klarinett" och "Klarinett i B" are the same instruments. This is also the case in the English texts I have read on instruments and instrumentation. Where is this distinction in English between "Horn in F" and "F Horn" documented?

**Comment by shoogle** [comment]
> In Swedish there is no such distinction

It's the same in English, at least colloquially, people will say "Horn in F" or "F Horn" interchangeably without stopping to think about whether they are talking about transposition or tuning.

This is more of a standard that we are trying to set rather than existing practice, so it's only documented [here](https://github.com/musescore/MuseScore/wiki/Translating-instrument-names).

**Comment by Magnus26** [comment]
Why not use strings 6130 and 6132?

**Comment by shoogle** [comment]
Actually, we *do* use those strings in the score, after you click "Done" in the New Score Dialog.

So the limitations only apply *within* the New Score Dialog, which doesn't matter as much as what you see in the score.

It may not stay that way forever though. But like I said before, you shouldn't judge the system based on the harmonicas.

**Comment by Magnus26** [comment]
> Actually, we _do_ use those strings in the score, after you click "Done" in the New Score Dialog.

Yes, I believe Joachim pointed that out earlier, but why not use them in the dialog window?

> So the limitations only apply _within_ the New Score Dialog, which doesn't matter as much as what you see in the score.

Being a word order sensitive Swede (the Swedish language is very dependent on word order and word compounds as a result of its bare bone grammar (or if it perhaps is the other way around)) the current situation is enough to cause a reaction.

> It may not stay that way forever though.

That sounds hopeful.

Happy midsummer, Peter!
